### ISSUE TITLE
Allow for query execution within range vector

### ISSUE THREAD

**Issue by Yentel** [prometheus/prometheus#8855]
## Proposal
**Use case. Why is this important?**
When trying to query the amount of bandwidth that has gone over an interface over the past 30 days, one can simply run this query (quick draft, only calculating In, skipping Out):
`increase(ifHCOutOctets{instance="<ip>",ifDescr="<ifdescr>}[30d])`

This works fine, however, when we want to alert when the amount of bandwidth has gone over X TB in the current month (1st day of the month 00:00:00 until now), one cannot use 30d but needs to enter the amount seconds since the start of the month.

For some reason, I cannot find any documentation nor do I succeed in replacing the range vector [30d] with a query that calculates the amount of seconds since this point in time. It seems that there is no room for executing a query within this range vector and only static values are allowed (Xs,Xm,Xh,Xd,...)

Using a recording rule that calculates the seconds_this_month, one would be in the possibility of using the vector [seconds_this_month]

Is there any chance this might be added in a future version or, if it really does exist, that this can be added to the documentation?
If there is no need for this method because there is another way of achieving this, please do let me know.

Thanks in advance!

**Comment by roidelapluie** [comment]
Yes there are changes that this is added in the future. We are looking for someone willing to put that in a design document, with all the options to achieve that and other things.

**Comment by Harkishen-Singh** [comment]
@roidelapluie what if we update the `[ ]` in PromQL to something like `[ start_time => end_time ]`. The `.*_time` here can be `unix seconds` or `RFC3339` and we fetch the data for the corresponding vector selector for that time range only.

**Comment by beorn7** [comment]
Hello from the bug scrub.

Note that some of this has been covered in https://docs.google.com/document/d/1jMeDsLvDfO92Qnry_JLAXalvMRzMSB1sBr9V7LolpYM/edit#heading=h.ormgh6nlwfxe , which has a lot of ideas that are already implemented by now. So definitely something we would like to tackle in the future.
